code/WorkInProgress/recycling/conveyor.dm

```diff
@@ -764,6 +970,38 @@ ADMIN_INTERACT_PROCS(/obj/machinery/conveyor_switch, proc/trigger)
 				C.setdir()
 		SEND_SIGNAL(src,COMSIG_MECHCOMP_TRANSMIT_SIGNAL,"switchTriggered")
 
+/obj/machinery/conveyor_switch/built/
+	desc = "A conveyor control switch. This one looks like it was built recently."
+
+
+	New()
+		. = ..()
+		UnsubscribeProcess()
+		START_TRACKING
+		UpdateIcon()
+		AddComponent(/datum/component/mechanics_holder)
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"trigger", PROC_REF(trigger))
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Refresh ID", PROC_REF(get_new_id))
+		src.get_new_id()
+
+	proc/get_new_id()
+		src.id = TIME // Create a new ID on the fly. It doesn't need to be readable, it just needs to be unique.
```

Comment:
this is not a good way to get a unique ID, I'd advise using ref instread

Reply:
Not sure what you mean by "ref" @TobleroneSwordfish

Reply:
Oh, wait, you mean the memory address for the atom? That makes sense.

Reply:
`ckey("\ref[src]")` or so - computers do this